code/modules/mob/living/simple_animal/hostile/terror_spiders/purple.dm

```diff
@@ -62,34 +62,54 @@
 			if(Q.stat == DEAD)
 				spider_myqueen = null
 				degenerate = 1
-				to_chat(src,"<span class='userdanger'>Your Queen has died! Her power no longer sustains you!</span>")
+				to_chat(src,"<span class='userdanger'>[Q] has died! Her power no longer sustains you!</span>")
 				return
 			queen_visible = 0
-			for(var/mob/living/M in view(src, vision_range))
+			for(var/mob/living/M in orange(src, vision_range))
 				if(M == Q)
 					queen_visible = 1
 					break
 			if(queen_visible)
 				cycles_noqueen = 0
 				if(spider_debug)
-					to_chat(src,"<span class='notice'>Queen visible.</span>")
+					to_chat(src,"<span class='notice'>[Q] visible.</span>")
 			else
 				cycles_noqueen++
 				if(spider_debug)
-					to_chat(src,"<span class='danger'>Queen NOT visible. Cycles: [cycles_noqueen].</span>")
-			if(cycles_noqueen == 3)
+					to_chat(src,"<span class='danger'>[Q] NOT visible. Cycles: [cycles_noqueen].</span>")
+			var/area/A = get_area(spider_myqueen)
+			if(cycles_noqueen == 6)
```

Comment:
Can use a switch for this.
cycles_noqueen is also quite abstract for what the variable actually is. It depicts the amount of minutes without having seen the queen.

Reply:
Its not minutes, its Life() cycles.

Reply:
it's not minutes no. Mistake on my part. It's in 10 seconds. Since every 5 ticks this one gets incremented.